Library/src/Multiplayer/SignalRMsgPackEntitySerialiser.cpp

```diff
@@ -27,6 +27,176 @@ namespace csp::multiplayer
 {
 using namespace msgpack_typeids;
 
+namespace
+{
+std::pair<ItemComponentData, signalr::value> ReplicatedValueToSignalRValue(const ReplicatedValue& Value)
+{
+	ItemComponentData ValueType;
+	signalr::value NewValue;
+
+	switch (Value.GetReplicatedValueType())
+	{
+		case ReplicatedValueType::Boolean:
+			ValueType = ItemComponentData::BOOL;
+			NewValue  = signalr::value(Value.GetBool());
+			break;
+		case ReplicatedValueType::Integer:
+			ValueType = ItemComponentData::INT64;
+			NewValue  = signalr::value(Value.GetInt());
+			break;
+		case ReplicatedValueType::Float:
+			ValueType = ItemComponentData::FLOAT;
+			NewValue  = signalr::value(Value.GetFloat());
+			break;
+		case ReplicatedValueType::String:
+			ValueType = ItemComponentData::STRING;
+			NewValue  = signalr::value(Value.GetString().c_str(), Value.GetString().Length());
+			break;
+		case ReplicatedValueType::Vector2:
+		{
+			ValueType	= ItemComponentData::FLOAT_ARRAY;
+			auto VValue = Value.GetVector2();
+			NewValue	= signalr::value(std::vector {signalr::value(VValue.X), signalr::value(VValue.Y)});
+			break;
+		}
+		case ReplicatedValueType::Vector3:
+		{
+			ValueType	= ItemComponentData::FLOAT_ARRAY;
+			auto VValue = Value.GetVector3();
+			NewValue	= signalr::value(std::vector {signalr::value(VValue.X), signalr::value(VValue.Y), signalr::value(VValue.Z)});
+			break;
+		}
+		case ReplicatedValueType::Vector4:
+		{
+			ValueType	= ItemComponentData::FLOAT_ARRAY;
+			auto VValue = Value.GetVector4();
+			NewValue	= signalr::value(
+				   std::vector {signalr::value(VValue.X), signalr::value(VValue.Y), signalr::value(VValue.Z), signalr::value(VValue.W)});
+			break;
+		}
+		case ReplicatedValueType::Map:
+		{
+			ValueType = ItemComponentData::GENERIC_MAP;
+
+			// Creates a custom signalr data structure to hold an array of key-value pairs
+			const auto& Map = Value.GetMap();
+			auto Keys		= Map.Keys();
```

Comment:
(very minor) your spaces are misaligned

Reply:
This is Github formatting. It looks correct on my end
![image](https://github.com/user-attachments/assets/dfb1c4d4-e46d-483e-9c70-080e65e7ad42)